antismash/common/secmet/test/test_locations.py

```diff
@@ -29,81 +29,117 @@ def setUp(self):
 
     def test_position_conversion_nonzero_start(self):
         location = FeatureLocation(6, 21, strand=1)
         assert len(location) == 15
-        assert self.func(0, 2, location) == (6, 12)
-        assert self.func(1, 4, location) == (9, 18)
+        assert self.func(0, 2, location, 1) == (6, 12)
+        assert self.func(1, 4, location, 1) == (9, 18)
+        assert self.func(0, 2, location, 2) == (7, 13) # /codon_start=2
+        assert self.func(1, 4, location, 2) == (10, 19) # /codon_start=2
+        assert self.func(0, 2, location, 3) == (8, 14) # /codon_start=3
+        assert self.func(1, 4, location, 3) == (11, 20) # /codon_start=3
 
         location = FeatureLocation(6, 21, strand=-1)
         assert len(location) == 15
-        assert self.func(0, 2, location) == (15, 21)
-        assert self.func(1, 4, location) == (9, 18)
+        assert self.func(0, 2, location, 1) == (15, 21)
+        assert self.func(1, 4, location, 1) == (9, 18)
+        assert self.func(0, 2, location, 2) == (14, 20) # /codon_start=2
+        assert self.func(1, 4, location, 2) == (8, 17) # /codon_start=2
+        assert self.func(0, 2, location, 3) == (13, 19) # /codon_start=3
+        assert self.func(1, 4, location, 3) == (7, 16) # /codon_start=3
 
     def test_position_conversion_nonzero_compound(self):
         location = CompoundLocation([FeatureLocation(6, 18, strand=1),
                                      FeatureLocation(24, 27, strand=1)])
         assert len(location) == 15
-        assert self.func(0, 2, location) == (6, 12)
-        assert self.func(1, 4, location) == (9, 18)
-        assert self.func(3, 5, location) == (15, 27)
+        assert self.func(0, 2, location, 1) == (6, 12)
+        assert self.func(1, 4, location, 1) == (9, 18)
+        assert self.func(3, 5, location, 1) == (15, 27)
 
         location = CompoundLocation([FeatureLocation(6, 15, strand=-1),
                                      FeatureLocation(21, 27, strand=-1)])
         assert len(location) == 15
-        assert self.func(0, 2, location) == (21, 27)
-        assert self.func(1, 4, location) == (9, 24)
-        assert self.func(3, 5, location) == (6, 12)
+        assert self.func(0, 2, location, 1) == (9, 15)
+        assert self.func(1, 4, location, 1) == (24, 12)
+        assert self.func(3, 5, location, 1) == (21, 27)
 
     def test_position_conversion_compound_forward(self):
         location = CompoundLocation([FeatureLocation(0, 6, strand=1),
                                      FeatureLocation(9, 18, strand=1)])
         assert len(location) == 15
-        assert self.func(0, 4, location) == (0, 15)
-        assert self.func(1, 5, location) == (3, 18)
+        assert self.func(0, 4, location, 1) == (0, 15)
+        assert self.func(0, 4, location, 2) == (1, 16) # /codon_start=2
+        assert self.func(0, 4, location, 3) == (2, 17) # /codon_start=3
+        assert self.func(1, 5, location, 1) == (3, 18)
```

Comment:
There's some suspiciously missing tests here. Where's the case of 1, 5 with codon starts of 2 or 3? Should we expect them to walk off the end of the location or should they raise an error?

Reply:
Definitely not walking out.